kt/entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/checks/FunctionArgCountCheck.kt

```diff
@@ -0,0 +1,17 @@
+package godot.entrygenerator.checks
+
+import godot.entrygenerator.model.SourceFile
+import godot.entrygenerator.utils.Logger
+
+class FunctionArgCountCheck(logger: Logger, sourceFiles: List<SourceFile>): BaseCheck(logger, sourceFiles) {
+    override fun execute() {
+        sourceFiles
+            .flatMap { it.registeredClasses }
+            .flatMap { it.functions }
+            .forEach { registeredFunction ->
+                if (registeredFunction.parameters.size > 5) {
+                    logger.error("RegisteredFunction ${registeredFunction.fqName} has more than 5 arguments. More than that is currently not supported as it's very inefficient to unpack that many values. If you need more, either wrap them in a wrapper object or pass a VariantArray containing your values.")
```

Comment:
I don't think it's accurate to say it's "very inefficient" to unpack many values. The cost is purely linear with the number of argument, it's not like jumping from 5 to 6 arguments would be a major cost increase.
I think the original reason why we limited to 5 arguments was because it was an actual limit in the Godot C++ code, but if I remember correctly that restriction is lifted in 4.0, you can basically use as many arguments as you want (Thanks to macro/template black magic).

Reply:
So maybe we should instead drop this limitation in our cpp code

Reply:
Yeah that would be good. But I'd do that in a separate PR